recipes/SEP-28k/train.py

```diff
@@ -0,0 +1,255 @@
+import os
+import sys
+from functools import partial
+
+import numpy as np
+import torch
+import torchaudio
+from eval_utils import my_confusion_matrix
+from hyperpyyaml import load_hyperpyyaml
+from sep28k_prepare import prepare_sep28k
+from torch.utils.tensorboard import SummaryWriter
+
+import speechbrain as sb
+from speechbrain.dataio.batch import PaddedBatch
+from speechbrain.utils import hpopt as hp
+
+os.environ["CUDA_LAUNCH_BLOCKING"] = "1"
+
+
+class SEP28kBrain(sb.Brain):
+    def compute_feats(self, wavs, lens, stage):
+        """Verify wavs length (although padding and lens handles it)"""
+        # All clips should be 16Khz and 3 seconds long thus size 48000
+        if wavs.shape[1] > 48000:
+            wavs = wavs[:, :48000]
+        elif wavs.shape[1] < 48000:
+            pad = torch.zeros([wavs.shape[0], 48000 - wavs.shape[1]]).to(
+                "cuda:0"
+            )
+            wavs = torch.cat([wavs, pad], dim=1)
+        return wavs
+
+    def compute_forward(self, batch, stage):
+        """Input waveform into the model and outputs binary classification"""
+        batch = batch.to(self.device)
+        waveforms, lens = batch.waveform
+        waveforms = self.compute_feats(waveforms, lens, stage)
+        bin_out = self.modules.model(waveforms)
+        return {"bin_pred": bin_out}
+
+    def compute_objectives(self, predictions, batch, stage):
+        """Computes the loss (Binary Cross Entropy) given predictions and targets."""
+        labels = batch.label.data
+        loss = sb.nnet.losses.bce_loss(
+            predictions["bin_pred"].squeeze(1).float(),
+            labels.squeeze(1).float(),
+            pos_weight=torch.Tensor([self.hparams.positive]).to("cuda:0"),
+        )
+        binary_preds = torch.round(
+            torch.sigmoid(predictions["bin_pred"])
+        )  # torch.argmax(, axis=1)
+        self.y_true_binary = torch.cat((self.y_true_binary, labels))
+        self.y_preds_binary = torch.cat((self.y_preds_binary, binary_preds))
+        return loss
+
+    def on_stage_start(self, stage, epoch):
+        "Gets called when a stage (either training, validation, test) starts."
+        self.y_preds_binary = torch.tensor(()).to("cuda:0")
+        self.y_true_binary = torch.tensor(()).to("cuda:0")
```

Comment:
We have [BinaryMetricStats](https://speechbrain.readthedocs.io/en/latest/API/speechbrain.utils.metric_stats.html#speechbrain.utils.metric_stats.BinaryMetricStats) which may be an easier way of doing what you're computing here, though I don't think it does confusion matrix yet. You could use the class to store the predictions and pass to scikit-learn to get confusion matrix.

Reply:
But if you keep it as-is, use `self.device` instead of `"cuda:0"`